code/API_definitions/device-swap.yaml

```diff
@@ -0,0 +1,399 @@
+
+      Which specific authorization flows are to be used will be determined during onboarding process, happening between the API Client and the Telco Operator exposing the API, taking into account the declared purpose for accessing the API, while also being subject to the prevailing legal framework dictated by local legislation.
+
+      It is important to remark that in cases where personal user data is processed by the API, and users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of 3-legged access tokens becomes mandatory. This measure ensures that the API remains in strict compliance with user privacy preferences and regulatory obligations, upholding the principles of transparency and user-centric data control.
+
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: wip
+  x-camara-commonalities: 0.4.0
+servers:
+  - url: '{apiRoot}/device-swap/vwip'
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root, defined by the service provider, e.g. `api.example.com` or `api.example.com/somepath`
+tags:
+  - name: Retrieve Device Swap Date
+    description: Receive the last date in which the device of the end-user was swapped
+  - name: Check Device Swap
+    description: Validate if the SIM of the end-used has been installed in a different device during a past period
+paths: 
+  /retrieve-date:
+    post:
+      tags:
+        - Retrieve Device Swap Date
+      description: Get timestamp of last device swap for a mobile user account provided with phone number.
+      operationId: retrieveDeviceSwapDate
+      security:
+        - openId:
+          - device-swap
+        - openId:
+          - device-swap:retrieve-date
+      parameters:
+        - $ref: '#/components/parameters/x-correlator'
+      requestBody:
+        description: |
+          Create a device swap date request for a phone number.
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CreateDeviceSwapDate"
+        required: true
+      responses:
+        "200":
+          description: Contains information about Device swap change
+          headers:
+            x-correlator:
+              $ref: '#/components/headers/x-correlator'
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/DeviceSwapInfo"
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/Generic404"
+        "429":
```

Comment:
See comment for 422 & 429 below as same apply here.

Reply:
both solved